zingo-testutils/src/lightclient.rs

```diff
@@ -0,0 +1,90 @@
+pub async fn get_base_address(client: &LightClient, pooltype: PoolType) -> String {
+    match pooltype {
+        PoolType::Transparent => client.do_addresses().await[0]["receivers"]["transparent"]
+            .clone()
+            .to_string(),
+        PoolType::Shielded(ShieldedProtocol::Sapling) => client.do_addresses().await[0]
+            ["receivers"]["sapling"]
+            .clone()
+            .to_string(),
+        PoolType::Shielded(ShieldedProtocol::Orchard) => {
+            client.do_addresses().await[0]["address"].take().to_string()
+        }
+    }
+}
+/// Helpers to provide raw_receivers to lightclients for send and shield, etc.
+pub mod from_inputs {
+    use zingolib::lightclient::LightClient;
+
+    /// Panics if the address, amount or memo conversion fails.
+    pub fn receivers_from_send_inputs(
+        raw_receivers: Vec<(&str, u64, Option<&str>)>,
+        chain: &zingoconfig::ChainType,
+    ) -> zingolib::data::receivers::Receivers {
+        raw_receivers
+            .into_iter()
+            .map(|(address, amount, memo)| {
+                let recipient_address =
+                    zingolib::utils::conversion::address_from_str(address, chain)
+                        .expect("should be a valid address");
+                let amount = zingolib::utils::conversion::zatoshis_from_u64(amount)
+                    .expect("should be inside the range of valid zatoshis");
+                let memo = memo.map(|memo| {
+                    zingolib::wallet::utils::interpret_memo_string(memo.to_string())
+                        .expect("should be able to interpret memo")
+                });
+
+                zingolib::data::receivers::Receiver::new(recipient_address, amount, memo)
+            })
+            .collect()
+    }
+    /// In a test give sender a raw_receiver to encode and send to
+    pub async fn send(
+        sender: &zingolib::lightclient::LightClient,
+        raw_receivers: Vec<(&str, u64, Option<&str>)>,
+    ) -> Result<String, String> {
+        let receivers = receivers_from_send_inputs(raw_receivers, &sender.config().chain);
+        sender.do_send(receivers).await.map(|txid| txid.to_string())
+    }
+    /// Panics if the address conversion fails.
+    pub async fn shield(
+        shielder: &LightClient,
+        pools_to_shield: &[zingolib::wallet::Pool],
+        address: Option<&str>,
+    ) -> Result<String, String> {
+        let address = address.map(|addr| {
+            zingolib::utils::conversion::address_from_str(addr, &shielder.config().chain)
+                .expect("should be a valid address")
+        });
+        shielder
+            .do_shield(pools_to_shield, address)
+            .await
+            .map(|txid| txid.to_string())
+    }
+
+    /// Creates a [`zcash_client_backend::zip321::TransactionRequest`] from rust primitives for simplified test writing.
+    pub fn transaction_request_from_send_inputs(
+        requester: &zingolib::lightclient::LightClient,
```

Comment:
maybe i prefer `sender` as its more clear to me but non-blocking. I'm pretty sure this previously had ChainType as the input which I liked because you shouldnt need to create an entire lightclient to create a transaction request in testing but I understand the organisation here and im not going to block on this either

Reply:
Happy to add that back in as needed.